app/models/dog.rb

```diff
@@ -7,4 +8,12 @@ class Dog < ActiveRecord::Base
     bucket: ENV['BUCKET_NAME']
 
   validates_attachment_content_type :img, :content_type => /\Aimage\/.*\Z/
+
+  def address
+    [street, city, state, zipcode].join(", ")
+  end
+
+  def location
+    Geokit::Geocoders::GoogleGeocoder.geocode(address, :bias => 'us')
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.